src/withPseudoState.js

```diff
@@ -87,9 +87,12 @@ function rewriteStyleSheets(shadowRoot) {
                   states.push(`.pseudo-${state}`)
                   return ""
                 })
-                const stateSelector = shadowRoot
-                  ? `:host(${states.join("")}) ${plainSelector}`
-                  : `${states.join("")} ${plainSelector}`
+                let stateSelector;
+                if (shadowRoot && states.length) {
+                  stateSelector = `:host(${states.join('')}) ${selectorText.startsWith(':host') ? '' : plainSelector}`;
```

Comment:
`selectorText` can contain several selectors (separated with a comma). Will `:host` only ever be on the first one? You should probably test against `selector` instead.

Reply:
Changed to `selector.includes(":host")`